sound/soc/sof/trace.c

```diff
@@ -321,6 +321,7 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 		count = avail;
 
 	/* copy available trace data to debugfs */
+	snd_dma_buffer_sync(&sdev->dmatb, SNDRV_DMA_SYNC_CPU);
```

Comment:
can you clarify why this is needed? I don't recall doing this for capture streams?

Reply:
@plbossart, for playback and capture it is done in sound/core/pcm_lib.c
https://github.com/thesofproject/linux/blob/9622109106be4259bd0eddc3eb566c7689246d58/sound/core/pcm_lib.c#L2276-L2283

In case  of  dtrace the `snd_dma_buffer_sync()` can be NOP or real sync depending on the platform.
!CONFIG_SND_DMA_SGBUF will result `dma_alloc_coherent()` allocation, not needing sync (snd_malloc_ops.sync is not set), but if CONFIG_SND_DMA_SGBUF set then `dma_need_sync()` will determine if a sync is needed or not.

This likely going to be NOP in most cases, but it is better to use proper logic to avoid surprises.

Reply:
Thanks @ujfalusi, very useful. I would add a comment along the lines of 'teh sdn_dma_buffer_sync() is already handled by the ALSA core, this patch only replicates the same behavior for the trace'.